pkg/scheduler/scheduler.go

```diff
@@ -319,6 +319,32 @@ func (s *Scheduler) nominate(ctx context.Context, workloads []workload.Info, sna
 	return entries
 }
 
+// resourcesToReserve calculates how much of available resources in cq assignment should be reserved.
+func resourcesToReserve(e *entry, cq *cache.ClusterQueue) cache.FlavorResourceQuantities {
+	if e.assignment.RepresentativeMode() != flavorassigner.Preempt {
+		return e.assignment.Usage
+	}
+	reservedUsage := make(cache.FlavorResourceQuantities)
+	for flavor, resourceUsage := range e.assignment.Usage {
+		cqQuota := make(map[corev1.ResourceName]*cache.ResourceQuota)
+		for _, rg := range cq.ResourceGroups {
```

Comment:
I think you can use `cq.RGByResource` to obtain the quotas for a resource, instead of having to loop over the resourceGroups.

https://github.com/kubernetes-sigs/kueue/blob/a7ad78db2ca32d2f10f1d8fff57c98f0d9b4a824/pkg/cache/clusterqueue.go#L47


There is this function which you can use to populate that map in unit tests https://github.com/kubernetes-sigs/kueue/blob/a7ad78db2ca32d2f10f1d8fff57c98f0d9b4a824/pkg/cache/clusterqueue.go#L257

but, in production, it is maintained by the cache and snapshot logic.